arduino/cores/packagemanager/loader.go

```diff
@@ -374,14 +374,47 @@ func (pm *PackageManager) loadBoards(platform *cores.PlatformRelease) error {
 
 	propertiesByBoard := boardsProperties.FirstLevelOf()
 
-	platform.Menus = propertiesByBoard["menu"]
+	if menus, ok := propertiesByBoard["menu"]; ok {
+		platform.Menus = menus
+	} else {
+		platform.Menus = properties.NewMap()
+	}
 
-	delete(propertiesByBoard, "menu") // TODO: check this one
+	// This is not a board id so we remove it to correctly
+	// set all other boards properties
+	delete(propertiesByBoard, "menu")
 
+	skippedBoards := []string{}
 	for boardID, boardProperties := range propertiesByBoard {
-		boardProperties.Set("_id", boardID) // TODO: What is that for??
-		board := platform.GetOrCreateBoard(boardID)
+		var board *cores.Board
+		for key := range boardProperties.AsMap() {
+			if !strings.HasPrefix(key, "menu") {
```

Comment:
This causes a panic if a board contains a property that starts with menu:
```
samd21_dev.menufoo=bar
```
```
$ ./arduino-cli core list
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/arduino/arduino-cli/arduino/cores/packagemanager.(*PackageManager).loadBoards(0xc000045200, 0xc0002b4fc0, 0xc00097133f, 0x7)
        C:/Users/per/go/src/github.com/arduino/arduino-cli/arduino/cores/packagemanager/loader.go:398 +0xb08
github.com/arduino/arduino-cli/arduino/cores/packagemanager.(*PackageManager).loadPlatformRelease(0xc000045200, 0xc0002b4fc0, 0xc0002b8410, 0x0, 0x0)
        C:/Users/per/go/src/github.com/arduino/arduino-cli/arduino/cores/packagemanager/loader.go:343 +0x7b2
github.com/arduino/arduino-cli/arduino/cores/packagemanager.(*PackageManager).loadPlatform(0xc000045200, 0xc000090a00, 0xc0002b83d0, 0x4)
        C:/Users/per/go/src/github.com/arduino/arduino-cli/arduino/cores/packagemanager/loader.go:281 +0x1189
github.com/arduino/arduino-cli/arduino/cores/packagemanager.(*PackageManager).loadPlatforms(0xc000045200, 0xc000090a00, 0xc0002b8330, 0x8, 0xc00012b290, 0x0)
        C:/Users/per/go/src/github.com/arduino/arduino-cli/arduino/cores/packagemanager/loader.go:173 +0x29e
github.com/arduino/arduino-cli/arduino/cores/packagemanager.(*PackageManager).LoadHardwareFromDirectory(0xc000045200, 0xc0000f1ef0, 0xc0002b7e00, 0xc0002b8210, 0xc0005eec50)
        C:/Users/per/go/src/github.com/arduino/arduino-cli/arduino/cores/packagemanager/loader.go:132 +0x657
github.com/arduino/arduino-cli/arduino/cores/packagemanager.(*PackageManager).LoadHardwareFromDirectories(0xc000045200, 0xc0005eec50, 0x1, 0x1, 0x7, 0x8, 0xc0001eb050)
        C:/Users/per/go/src/github.com/arduino/arduino-cli/arduino/cores/packagemanager/loader.go:52 +0x92
github.com/arduino/arduino-cli/arduino/cores/packagemanager.(*PackageManager).LoadHardware(0xc000045200, 0xc0005ebcb0, 0x0, 0x0)
        C:/Users/per/go/src/github.com/arduino/arduino-cli/arduino/cores/packagemanager/loader.go:37 +0x57
github.com/arduino/arduino-cli/commands.Init(0xc00094dc10, 0xc000523bf0, 0xc0000ac150)
        C:/Users/per/go/src/github.com/arduino/arduino-cli/commands/instances.go:228 +0x37e
github.com/arduino/arduino-cli/cli/instance.Init(0xc0000ac150, 0xc0000ac120, 0x0, 0xc00094dcd0)
        C:/Users/per/go/src/github.com/arduino/arduino-cli/cli/instance/instance.go:74 +0x1af
github.com/arduino/arduino-cli/cli/instance.CreateAndInit(0xa)
        C:/Users/per/go/src/github.com/arduino/arduino-cli/cli/instance/instance.go:43 +0x92
github.com/arduino/arduino-cli/cli/core.runListCommand(0xc000013340, 0x20bee18, 0x0, 0x0)
        C:/Users/per/go/src/github.com/arduino/arduino-cli/cli/core/list.go:52 +0x3b
github.com/spf13/cobra.(*Command).execute(0xc000013340, 0x20bee18, 0x0, 0x0, 0xc000013340, 0x20bee18)
        C:/Users/per/go/pkg/mod/github.com/spf13/cobra@v1.0.1-0.20200710201246-675ae5f5a98c/command.go:846 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0xc0005fe580, 0x43, 0xc0005a77a0, 0xc000986460)
        C:/Users/per/go/pkg/mod/github.com/spf13/cobra@v1.0.1-0.20200710201246-675ae5f5a98c/command.go:950 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        C:/Users/per/go/pkg/mod/github.com/spf13/cobra@v1.0.1-0.20200710201246-675ae5f5a98c/command.go:887
main.main()
        C:/Users/per/go/src/github.com/arduino/arduino-cli/main.go:31 +0x8f
```
I guess it should be:
```suggestion
			if !strings.HasPrefix(key, "menu.") {
```

Reply:
Uuuh yeah, didn't consider that. :facepalm:

Reply:
Fixed in https://github.com/arduino/arduino-cli/pull/1326/commits/e0feac8f02fa00963301972bce53a3dc4175093c
I updated the testing `boards.txt` to verify it works too.